### PR TITLE
Create Modal Component

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

File: src/root/views/GoUI/components/Modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+  const hasHeader = !!title || withCloseButton;
+  const sizeStyle = styles[`size-${size}`];
+
+  const handleClickOutside = useCallback(() => {
+    if (closeOnClickOutside) {
+      onCloseButtonClick();
+    }
+  }, [onCloseButtonClick, closeOnClickOutside]);
+
```
Comment (shreeyash07) [inline src/root/views/GoUI/components/Modal/index.tsx:57]: Remove excessive line

File: src/root/views/GoUI/components/Modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  const handleEscape = useCallback(() => {
+    if (closeOnEscape) {
+      onCloseButtonClick();
+    }
+  }, [onCloseButtonClick, closeOnEscape]);
+
+  return (
+    <>
+      {opened && (
+        <Portal>
```
Comment (shreeyash07) [inline src/root/views/GoUI/components/Modal/index.tsx:68]: No need to wrap with portal

File: src/root/views/GoUI/examples/Modals/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.modals {
+  display: flex;
+  flex-direction: column;
```
Comment (shreeyash07) [inline src/root/views/GoUI/examples/Modals/styles.module.scss:3]: Please add padding like other examples

File: src/root/views/GoUI/components/Modal/index.tsx (hunk 1 above)
Comment (shreeyash07) [inline src/root/views/GoUI/components/Modal/index.tsx:57]: Remove excessive line

File: src/root/views/GoUI/components/Modal/index.tsx (hunk 2 above)
Comment (shreeyash07) [inline src/root/views/GoUI/components/Modal/index.tsx:68]: No need to wrap with portal

File: src/root/views/GoUI/examples/Modals/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  gap: var(--go-ui-spacing-md);
```
Comment (shreeyash07) [inline src/root/views/GoUI/examples/Modals/styles.module.scss:4]: Please add padding like other examples

**Review by shreeyash07 — CHANGES_REQUESTED** [review]

**Review by priyesh777 — APPROVED** [review]

File: src/root/views/GoUI/components/Modal/index.tsx (hunk 2 above)
Comment (samshara) [reply src/root/views/GoUI/components/Modal/index.tsx:68]: nice catch

**Review by shreeyash07 — APPROVED** [review]

**Pull request merged**